assets/src/scripts/components/hydrograph/parameters.js

```diff
@@ -64,12 +65,44 @@ export const addSparkLine = function(svgSelection, {seriesLineSegments, scales})
         .y(function(d) {
             return scales.y(d.value);
         });
-
-    for (const lineSegment of seriesLineSegments) {
-        if (lineSegment.classes.dataMask === null) {
-            svgSelection.append('path')
-                .attr('d', spark(lineSegment.points))
-                .attr('class', 'spark-line');
+    const seriesDataMasks = seriesLineSegments.map(x => x.classes.dataMask);
+    if (seriesDataMasks.includes(null)) {
+        for (const lineSegment of seriesLineSegments) {
+            if (lineSegment.classes.dataMask === null) {
+                svgSelection.append('path')
+                    .attr('d', spark(lineSegment.points))
+                    .attr('class', 'spark-line');
+            }
+        }
+    } else {
+        const centerElement = function (svgElement) {
+            const elementWidth = svgElement.node().getBoundingClientRect().width;
+            const xLocation = (SPARK_LINE_DIM.width - elementWidth) / 2;
+            svgElement.attr('x', xLocation);
+        };
+        let svgText = svgSelection.append('text')
+            .attr('x', 0)
+            .attr('y', 0)
+            .classed('sparkline-text', true);
+        const maskDescs = seriesDataMasks.map(x => MASK_DESC[x.toLowerCase()]);
+        const maskDesc = maskDescs.length === 1 ? maskDescs[0] : 'Masked';
+        const maskDescWords = maskDesc.split(' ');
+
+        if (maskDescWords.length > 1) {
+            Array.from(maskDescWords.entries()).forEach(x => {
+                const yPosition = 15 + x[0]*12;
+                const maskText = x[1];
+                let tspan = svgText.append('tspan')
+                    .attr('x', 0)
+                    .attr('y', yPosition)
+                    .text(maskText);
+                centerElement(svgText);
+                centerElement(tspan);
+            });
+        } else {
+            svgText.text(maskDesc)
+                .attr('y', '20');
+            centerElement(svgText);
```

Comment:
This looks good, so not suggesting we change it.. but just going through this with the tooltips, I think putting the text straight into the table as a div (or whatever) rather than drawing SVG could be simpler.

Reply:
I think I might keep it as is for now. I kind of like how the text is currently confined to the same space as the other sparklines... seems visually consistent.